src/test/java/games/strategy/triplea/ui/TimespanDialogTest.java

```diff
@@ -0,0 +1,39 @@
+package games.strategy.triplea.ui;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Instant;
+
+import org.junit.Test;
+
+import games.strategy.triplea.ui.TimespanDialog.TimeUnit;
+
+public class TimespanDialogTest {
+
+  @Test
+  public void testForeverReturnsNull() {
+    assertNull(TimeUnit.FOREVER.getInstant((int) (Math.random() * Integer.MAX_VALUE)));
+  }
+
+  @Test
+  public void testIntervalSizesBecomeBigger() {
+    final Instant minute = TimeUnit.MINUTES.getInstant(1);
+    final Instant hour = TimeUnit.HOURS.getInstant(1);
+    final Instant day = TimeUnit.DAYS.getInstant(1);
+    final Instant week = TimeUnit.WEEKS.getInstant(1);
+    final Instant month = TimeUnit.MONTHS.getInstant(1);
+    final Instant year = TimeUnit.YEARS.getInstant(1);
+
+    assertTrue(minute.isBefore(hour));
+    assertTrue(hour.isBefore(day));
+    assertTrue(day.isBefore(week));
+    assertTrue(week.isBefore(month));
+    assertTrue(month.isBefore(year));
+  }
+
+  @Test
+  public void testPositiveIntIsInFuture() {
+    assertTrue(Instant.now().isBefore(TimeUnit.MINUTES.getInstant(1)));
+  }
+}
```

Comment:
If you want to get some additional test coverage out of the logic in `prompt()`, you could extract the last few lines of that method so you could test the interaction with the `Consumer`:

* Verify consumer is not called when dialog is cancelled.
* Verify consumer receives `null` when expiration time is `null`.
* Verify consumer receives the correct `Date` (or just a non-`null` value) when expiration time is non-`null`.

The extraction would look something like:

```java
diff --git a/src/main/java/games/strategy/triplea/ui/TimespanDialog.java b/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
index 92cfc30..d17d10f 100644
--- a/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
+++ b/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
@@ -78,8 +78,13 @@ public class TimespanDialog {
             .add(comboBox)
             .build())
         .build(), title, JOptionPane.OK_CANCEL_OPTION);
+    runAction(action, returnType, (TimeUnit) comboBox.getSelectedItem(), (Integer) spinner.getValue());
+  }
+
+  @VisibleForTesting
+  void runAction(final Consumer<Date> action, final int returnType, final TimeUnit timeUnit, final Integer duration) {
     if (returnType == JOptionPane.OK_OPTION) {
-      final Instant instant = ((TimeUnit) comboBox.getSelectedItem()).getInstant((Integer) spinner.getValue());
+      final Instant instant = timeUnit.getInstant(duration);
       action.accept(instant == null ? null : Date.from(instant));
     }
   }
```